src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -1468,7 +1468,7 @@ internal override IEnumerable<CSharpAttributeData> GetCustomAttributesToEmit(Mod
 
         public static TypeSymbol TransformToTupleIfCompatible(TypeSymbol target)
         {
-            if (target.IsTupleCompatible())
+            if (target.IsTupleCompatible() && !target.IsErrorType())
```

Comment:
Should we check if the target is an error type first? Checking for tuple compatibility is slightly more expensive, I think.